modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -48,5 +59,67 @@ object ConstructionService {
 
       NonEmptyList.fromList(accountIdentifiers)
     }
+
+    def transactionToOperations(transaction: Transaction): Either[String, NonEmptyList[Operation]] =
+      refineV[Not[Equal[0L]]](transaction.amount.value.value).flatMap { amount =>
+        val operations: Either[String, List[Operation]] = refineV[Not[Equal[0L]]](-1L * transaction.amount.value.value).map { negAmount =>
+          val negativeOperation = Operation(
+            OperationIdentifier(OperationIndex(0L)),
+            none,
+            OperationType.Transfer,
+            none,
+            AccountIdentifier(transaction.destination, none).some,
+            Amount(AmountValue(negAmount), DAG).some
+          )
+
+          val positiveOperation = Operation(
+            OperationIdentifier(OperationIndex(1L)),
+            none,
+            OperationType.Transfer,
+            none,
+            AccountIdentifier(transaction.destination, none).some,
+            Amount(AmountValue(amount), DAG).some
+          )
+
+          List(negativeOperation, positiveOperation)
+        }
+
+        operations.flatMap { operationList =>
+          NonEmptyList.fromList(operationList) match {
+            case Some(value) => value.asRight
+            case None        => "No valid operations found.".asLeft
+          }
+        }
+      }
```

Comment:
You can use `for-comprehension` to get rid of nested code.

Also `NonEmptyList.fromList(List(negativeOperation, positiveOperation))` can be written as `NonEmptyList.of(negativeOperation, positiveOperation)` because you already know that it will pass.

Reply:
```scala
    def transactionToOperations(transaction: Transaction): Either[Throwable, NonEmptyList[Operation]] = {
      type Refinement = Not[Equal[0L]]

      val operations: Either[String, NonEmptyList[Operation]] = for {
        amount <- refineV[Refinement](transaction.amount.value.value)
        negAmount <- refineV[Refinement](-1L * transaction.amount.value.value)
        negativeOperation = Operation(
          OperationIdentifier(OperationIndex(0L)),
          none,
          OperationType.Transfer,
          none,
          AccountIdentifier(transaction.destination, none).some,
          Amount(AmountValue(negAmount), DAG).some
        )
        positiveOperation = Operation(
          OperationIdentifier(OperationIndex(1L)),
          none,
          OperationType.Transfer,
          none,
          AccountIdentifier(transaction.destination, none).some,
          Amount(AmountValue(amount), DAG).some
        )

        ops <- NonEmptyList.of(negativeOperation, positiveOperation).asRight
      } yield ops

      operations.leftMap(e => new Throwable(e))
    }
```

Reply:
Changed to using `NonEmptyList.of` and `for-comprehension`.